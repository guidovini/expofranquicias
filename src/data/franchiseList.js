const franchises = [
  {
    name: "Ecuafranquicias",
    category: "Consultoria",
    description: "Esta es la descripcion de Ecuafranquicias",
    price: 12000,
  },
  {
    name: "AEFRAN",
    category: "Organizacion",
    description: "Esta es la descripcion de AEFRAN",
    price: 15000,
  },
  {
    name: "Villagran Lara",
    category: "Consultoria",
    description: "Esta es la descripcion de Villagran Lara",
    price: 20000,
  },
  {
    name: "Mongos",
    category: "Restaurant",
    description: "Esta es la descripcion de Mongos",
    price: 45000,
  },
]

export default franchises
