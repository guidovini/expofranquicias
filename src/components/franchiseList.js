import React, { Component } from 'react'

import FranchiseItem from './franchiseItem'
import ControlBar from './controlBar'

import { franchises, countryList } from '../data/franchiseList'
import franchiseListStyles from './franchiseList.module.scss'

class FranchiseList extends Component {
  state = {
    orderBy: 'alpha',
    filterBy: 'all',
    searchBy: '',
  }

  handleOrder = val => {
    this.setState({
      orderBy: val,
    })
  }

  handleFilter = val => {
    this.setState({
      filterBy: val,
    })
  }

  handleSearch = val => {
    this.setState({
      searchBy: val,
    })
  }

  resetControlBar = () => {
    // this.setState({ orderBy: 'alpha', filterBy: 'all', searchBy: '' })
    this.setState({ searchBy: '' })
  }

  franchiseSelector = (franchises, orderBy, filterBy, searchBy) => {
    return franchises
      .filter(franchise => {
        const searchText = searchBy.toLowerCase().trim()
        return (
          franchise.name.toLowerCase().includes(searchText) ||
          franchise.concept.toLowerCase().includes(searchText)
        )
      })
      .filter(franchise => {
        if (filterBy === 'all') {
          return franchise
        }
        if (franchise.countryOrigin === filterBy) {
          return franchise.countryOrigin === filterBy
        }
      })
      .sort((a, b) => {
        const x = a.name.toLowerCase()
        const y = b.name.toLowerCase()
        if (orderBy === 'alpha') {
          if (x < y) {
            return -1
          }
          if (x > y) {
            return 1
          }
          return 0
        }

        if (orderBy === 'lower-price') return a.investment - b.investment
        if (orderBy === 'higher-price') return b.investment - a.investment
      })
  }

  render() {
    const { orderBy, filterBy, searchBy } = this.state
    const newFranchises = this.franchiseSelector(
      franchises,
      orderBy,
      filterBy,
      searchBy
    )

    return (
      <div>
        <div
          id="top-of-site-pixel-anchor"
          style={{
            position: 'absolute',
            width: '1px',
            height: '1px',
            // IF ACTIVATE CAUSES AN ERROR
            // top: '20rem',
            // top: '120vh',
            // top: '20rem',
            left: '0',
          }}
        ></div>
        <ControlBar
          countries={countryList}
          handleOrder={this.handleOrder}
          handleFilter={this.handleFilter}
          handleSearch={this.handleSearch}
          resetControlBar={this.resetControlBar}
          orderBy={orderBy}
          filterBy={filterBy}
          searchBy={searchBy}
        />
        <div className={franchiseListStyles.column}>
          {newFranchises.map(franchise => {
            return (
              franchise.investment && (
                <FranchiseItem
                  franchise={franchise}
                  key={`${franchise.name} + ${franchise.investment}`}
                />
              )
            )
          })}
        </div>
      </div>
    )
  }
}

export default FranchiseList
