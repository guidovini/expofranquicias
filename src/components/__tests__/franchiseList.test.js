import React from 'react'
import { render } from '@testing-library/react'
import { StaticQuery } from 'gatsby' // mocked

import { PureFranchiseList as FranchiseList } from '../franchiseList'

// Created Using the query from FranchiseList.js
const queryData = [
  {
    id: '1e773f6c-4336-5af2-a42c-b7f51c11affb',
    nombre: 'Autoseguro',
    categoria:
      'Comercializacion de Productos;Locales Comerciales;Productos de Venta',
    concepto: 'Comercialización de Purificadores de Agua',
    aniosOperacion: 37,
    numeroLocales: null,
    tamanoLocal: '6 - 16 m2',
    disponibilidadTerritorial: 'Todo el Ecuador',
    paisOrigen: 'Estados Unidos',
    tiempoRetornoInversion: '< 6 meses',
    inversion: 27500,
  },
  {
    id: 'e7f4efb1-7ac7-5a9a-8048-03163ccb290b',
    nombre: 'Test franchise',
    categoria: 'Restaurante;Bar;Entretenimiento',
    concepto: 'Resto - Bar - Karaoke',
    aniosOperacion: 17,
    numeroLocales: 11,
    tamanoLocal: '200 - 250 m2',
    disponibilidadTerritorial: 'Todo el Ecuador',
    paisOrigen: 'Ecuador',
    tiempoRetornoInversion: '2 años',
    inversion: 900000,
  },
  {
    id: 'e7f4efb1-7ac7-5a9a-8048-03163ccb290b',
    nombre: 'No investment franchise',
    categoria: 'Restaurante;Bar;Entretenimiento',
    concepto: 'Resto - Bar - Karaoke',
    aniosOperacion: 17,
    numeroLocales: 11,
    tamanoLocal: '200 - 250 m2',
    disponibilidadTerritorial: 'Todo el Ecuador',
    paisOrigen: 'Ecuador',
    tiempoRetornoInversion: '2 años',
    inversion: undefined,
  },
]
beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      site: {
        siteMetadata: {
          title: `Default Starter`,
        },
      },
      allImageSharp: {
        edges: [
          {
            node: {
              fluid: {
                base64: `pretend-i-am-a-base64-encoded-image`,
                aspectRatio: 1,
                src: 'this is the src',
                srcSet: `asdfasdf`,
                sizes: `100 200 300`,
                originalName: 'Autoseguro.jpg',
              },
            },
          },
        ],
      },
    })
  ).mockImplementationOnce(({ render }) =>
    render({
      allImageSharp: {
        edges: [
          {
            node: {
              fluid: {
                base64: `pretend-i-am-a-base64-encoded-image`,
                aspectRatio: 1,
                src: 'this is the src',
                srcSet: `asdfasdf`,
                sizes: `100 200 300`,
                originalName: 'Autoseguro.jpg',
              },
            },
          },
        ],
      },
    })
  )
})

describe('Franchise List', () => {
  test('Should_RenderFranchiseListCorrectly', () => {
    const { container } = render(<FranchiseList queryData={queryData} />)
    expect(container).toMatchSnapshot()
  })

  test('Should_RenderFranchiseList', () => {
    const { getByTestId } = render(<FranchiseList queryData={queryData} />)
    expect(getByTestId('franchise-list')).toBeInTheDocument()
  })

  test('Should_RenderFranchiseListItem', () => {
    const { getByText } = render(<FranchiseList queryData={queryData} />)
    expect(getByText('Test franchise')).toBeInTheDocument()
  })

  test('ShouldNot_RenderFranchiseItem_When_NoFranchiseInvestment', () => {
    const { queryByText } = render(<FranchiseList queryData={queryData} />)
    expect(queryByText('No investment franchise')).not.toBeInTheDocument()
  })

  test('Should_RenderMessage_When_NoData', () => {
    const { getByTestId } = render(<FranchiseList />)
    expect(getByTestId('franchise-list-no-data')).toBeInTheDocument()
  })

  test('Should_RenderControlBarCorrectly', () => {
    const { getByTestId } = render(<FranchiseList queryData={queryData} />)
    expect(getByTestId('controlBar')).toBeInTheDocument()
  })
})
