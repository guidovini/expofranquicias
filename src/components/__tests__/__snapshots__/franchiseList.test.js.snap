// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Franchise List Should_RenderFranchiseListCorrectly 1`] = `
<div>
  <div
    data-testid="franchise-list"
  >
    <div
      class="control-bar"
      data-testid="controlBar"
      id="myHeader"
    >
      <div
        class="field is-horizontal"
      >
        <div
          class="field-label is-normal"
        >
          <label
            class="label"
          >
            Ordenar:
          </label>
        </div>
        <div
          class="control"
        >
          <div
            class="select"
          >
            <select>
              <option
                value="alpha"
              >
                Alfabeticamente
              </option>
              <option
                value="lower-price"
              >
                Menor Precio
              </option>
              <option
                value="higher-price"
              >
                Mayor Precio
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        class="field is-horizontal"
      >
        <div
          class="field-label is-normal"
        >
          <label
            class="label"
          >
            Filtro por país:
          </label>
        </div>
        <div
          class="control has-icons-left"
        >
          <span
            class="select"
          >
            <select
              class="select"
            >
              <option
                value="all"
              >
                Todos
              </option>
              <option
                value="Ecuador"
              >
                Ecuador
              </option>
              <option
                value="Estados Unidos"
              >
                Estados Unidos
              </option>
            </select>
          </span>
          <span
            class="icon is-small is-left"
          >
            <i
              class="fas fa-globe"
            />
          </span>
        </div>
      </div>
      <div
        class="field is-horizontal"
      >
        <div
          class="field-label is-normal"
        >
          <label
            class="label"
          >
            Buscar:
          </label>
        </div>
        <div
          class="control is-expandable"
          style="flex-basis: 28rem;"
        >
          <input
            class="input"
            placeholder="Por nombre de franquicia o concepto de negocio"
            type="text"
            value=""
          />
        </div>
        <button
          class="buttonTrash"
          title="Reiniciar opciones"
        >
          <span
            class="icon"
          >
            <i
              class="fas fa-lg fa-trash"
            />
          </span>
        </button>
      </div>
    </div>
    <div
      class="column"
      id="stickyContent"
    >
      <div
        class="container"
      >
        <div
          class="imageBox"
          data-testid="franchise-logo"
        >
          <div
            class=" gatsby-image-wrapper"
            style="position: relative; overflow: hidden;"
          >
            <div
              style="width: 100%; padding-bottom: 100%;"
            />
            <img
              alt=""
              src="pretend-i-am-a-base64-encoded-image"
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
            />
            <picture>
              <img
                alt=""
                loading="lazy"
                sizes="100 200 300"
                src="this is the src"
                srcset="asdfasdf"
                style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
              />
            </picture>
            <noscript>
              &lt;picture&gt;&lt;img loading="lazy" sizes="100 200 300" srcset="asdfasdf" src="this is the src" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
            </noscript>
          </div>
        </div>
        <div
          class="data"
        >
          <h3
            class="title is-4 has-text-grey-dark"
            data-testid="franchise-name"
          >
            Autoseguro
          </h3>
          <p
            class="subtitle is-6 has-text-grey-dark"
            data-testid="franchise-concept"
            style="font-size: 1.125rem; opacity: 0.7;"
          >
            Comercialización de Purificadores de Agua
          </p>
          <div
            class="iconData"
          >
            <div
              class="iconRow"
              data-testid="franchise-years"
            >
              <span
                class="icon is-medium"
                style="color: rgb(170, 170, 170); margin-right: 0.4rem;"
              >
                <i
                  class="fas fa-lg fa-business-time"
                />
              </span>
              <span
                class="fieldContent"
                style="margin-right: 0.3em;"
              >
                37
              </span>
              <span
                class="fieldTitle"
                style="margin-right: 1.5em;"
              >
                 
                años de operación
              </span>
            </div>
            <div
              class="iconRow"
              data-testid="franchise-size"
            >
              <span
                class="icon is-medium"
                style="color: rgb(170, 170, 170); margin-right: 0.4rem;"
              >
                <i
                  class="fas fa-lg fa-ruler"
                />
              </span>
              <span
                class="fieldContent"
                style="margin-right: 0.3rem;"
              >
                6 - 16 
              </span>
              <span
                class="fieldTitle"
                style="margin-right: 0.65rem;"
              >
                m
                <sup>
                  2
                </sup>
              </span>
            </div>
          </div>
          <div
            class="labelData"
          >
            <div
              data-testid="franchise-territory"
            >
              <div
                class="field"
              >
                <h4
                  class="fieldTitle"
                >
                  Disponibilidad:
                </h4>
                <p
                  class="fieldContent"
                >
                  Todo el Ecuador
                </p>
              </div>
            </div>
            <div
              data-testid="franchise-countryOrigin"
            >
              <div
                class="field"
              >
                <h4
                  class="fieldTitle"
                >
                  País de Origen:
                </h4>
                <img
                  alt="Estados Unidos flag"
                  class="flag"
                  src="test-file-stub"
                  title="Estados Unidos"
                />
              </div>
            </div>
            <div
              data-testid="franchise-roi"
            >
              <div
                class="field"
              >
                <h4
                  class="fieldTitle"
                >
                  Tiempo ROI:
                </h4>
                <p
                  class="fieldContent"
                >
                  &lt; 6 meses
                </p>
              </div>
            </div>
          </div>
          <div
            class="field"
            data-testid="franchise-investment"
            style="flex-direction: column; justify-content: flex-end; align-content: flex-end; flex-grow: 1;"
          >
            <h4
              class="priceTitle"
            >
              Inversión Aproximada Desde
            </h4>
            <p
              class="price"
            >
              $ 27,500
            </p>
          </div>
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="imageBox"
          data-testid="franchise-logo"
        >
          <div
            class=" gatsby-image-wrapper"
            style="position: relative; overflow: hidden;"
          >
            <div
              style="width: 100%; padding-bottom: 100%;"
            />
            <img
              alt=""
              src="pretend-i-am-a-base64-encoded-image"
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
            />
            <picture>
              <img
                alt=""
                loading="lazy"
                sizes="100 200 300"
                src="this is the src"
                srcset="asdfasdf"
                style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
              />
            </picture>
            <noscript>
              &lt;picture&gt;&lt;img loading="lazy" sizes="100 200 300" srcset="asdfasdf" src="this is the src" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
            </noscript>
          </div>
        </div>
        <div
          class="data"
        >
          <h3
            class="title is-4 has-text-grey-dark"
            data-testid="franchise-name"
          >
            Test franchise
          </h3>
          <p
            class="subtitle is-6 has-text-grey-dark"
            data-testid="franchise-concept"
            style="font-size: 1.125rem; opacity: 0.7;"
          >
            Resto - Bar - Karaoke
          </p>
          <div
            class="iconData"
          >
            <div
              class="iconRow"
              data-testid="franchise-years"
            >
              <span
                class="icon is-medium"
                style="color: rgb(170, 170, 170); margin-right: 0.4rem;"
              >
                <i
                  class="fas fa-lg fa-business-time"
                />
              </span>
              <span
                class="fieldContent"
                style="margin-right: 0.3em;"
              >
                17
              </span>
              <span
                class="fieldTitle"
                style="margin-right: 1.5em;"
              >
                 
                años de operación
              </span>
            </div>
            <div
              class="iconRow"
              data-testid="franchise-locations"
            >
              <span
                class="icon is-medium"
                style="color: rgb(170, 170, 170); margin-right: 0.4rem;"
              >
                <i
                  class="fas fa-lg fa-store"
                />
              </span>
              <span
                class="fieldContent"
                style="margin-right: 0.3em;"
              >
                11
              </span>
              <span
                class="fieldTitle"
                style="margin-right: 1.5em;"
              >
                 
                locales
              </span>
            </div>
            <div
              class="iconRow"
              data-testid="franchise-size"
            >
              <span
                class="icon is-medium"
                style="color: rgb(170, 170, 170); margin-right: 0.4rem;"
              >
                <i
                  class="fas fa-lg fa-ruler"
                />
              </span>
              <span
                class="fieldContent"
                style="margin-right: 0.3rem;"
              >
                200 - 250 
              </span>
              <span
                class="fieldTitle"
                style="margin-right: 0.65rem;"
              >
                m
                <sup>
                  2
                </sup>
              </span>
            </div>
          </div>
          <div
            class="labelData"
          >
            <div
              data-testid="franchise-territory"
            >
              <div
                class="field"
              >
                <h4
                  class="fieldTitle"
                >
                  Disponibilidad:
                </h4>
                <p
                  class="fieldContent"
                >
                  Todo el Ecuador
                </p>
              </div>
            </div>
            <div
              data-testid="franchise-countryOrigin"
            >
              <div
                class="field"
              >
                <h4
                  class="fieldTitle"
                >
                  País de Origen:
                </h4>
                <img
                  alt="Ecuador flag"
                  class="flag"
                  src="test-file-stub"
                  title="Ecuador"
                />
              </div>
            </div>
            <div
              data-testid="franchise-roi"
            >
              <div
                class="field"
              >
                <h4
                  class="fieldTitle"
                >
                  Tiempo ROI:
                </h4>
                <p
                  class="fieldContent"
                >
                  2 años
                </p>
              </div>
            </div>
          </div>
          <div
            class="field"
            data-testid="franchise-investment"
            style="flex-direction: column; justify-content: flex-end; align-content: flex-end; flex-grow: 1;"
          >
            <h4
              class="priceTitle"
            >
              Inversión Aproximada Desde
            </h4>
            <p
              class="price"
            >
              $ 900,000
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
